taskrunner/host_gateway.go

```diff
@@ -168,6 +163,12 @@ func (agbh *AgentHostGateway) prepRemoteHost(hostObj host.Host, sshOptions []str
 	// run the setup script with the agent
 	if logs, err := hostutil.RunSSHCommand("setup", hostutil.SetupCommand(&hostObj), sshOptions, hostObj); err != nil {
 		event.LogProvisionFailed(hostObj.Id, logs)
+		grip.Errorf("error running setup script on %s", hostObj.Id)
+		// there is no guarantee setup scripts are idempotent, so we terminate the host if the setup script fails
+		if err := hostObj.Terminate(); err != nil {
```

Comment:
can we call "DisablePosionedHost" rather than Terminate (terminate changes state in the db but doesn't enguage the cloud manager.

Reply:
done